### PR TITLE
Test without windows.h

### PR DESCRIPTION
**Pull request by gerritvanos**
added OS specific part in native implementation,
Also changed makefile to work with travis and don't ignore error while using run command

### CODE REVIEW

File: libraries/hwlib/library/targets/hwlib-native.hpp [hunk 1]
```diff
@@ -13,24 +13,50 @@
...
 
 #ifdef _HWLIB_ONCE 
 
 uint64_t now_ticks(){
    // https://stackoverflow.com/questions/1695288/getting-the-current-time-in-milliseconds-from-the-system-clock-in-windows	 
-   FILETIME ft_now;
-   GetSystemTimeAsFileTime( &ft_now );
-   uint64_t ll_now = (LONGLONG)ft_now.dwLowDateTime + ((LONGLONG)(ft_now.dwHighDateTime) << 32LL);   
-   return ll_now / 10;
+   #ifdef __unix__
```
Comment (itzandroidtab) [inline libraries/hwlib/library/targets/hwlib-native.hpp]: This looks to be indented a bit strange

File: programs/bmptk/Makefile.inc [hunk 2]
```diff
@@ -427,7 +427,7 @@ define Windows
    PREFIX            := $(GCC-WIN)/bin/
    TARGET_CPP_FLAGS  += -mno-ms-bitfields   
    TARGET_C_FLAGS    += -mno-ms-bitfields   
-   TARGET_LN_FLAGS   += -lgdi32
+   #TARGET_LN_FLAGS   += -lgdi32
```
Comment (itzandroidtab) [inline programs/bmptk/Makefile.inc]: Is there a reason to disable the windows graphics device interface [(GDI)](https://docs.microsoft.com/en-us/windows/desktop/gdi/windows-gdi) on windows machines?

File: libraries/hwlib/library/targets/hwlib-native.hpp [hunk 3]
```diff
@@ -13,24 +13,50 @@
 // this file contains Doxygen lines
 /// @file
 
+#ifdef __unix__                    /* __unix__ is usually defined by compilers targeting Unix systems */
+
+    #define OS_Windows 0
```
Comment (itzandroidtab) [inline libraries/hwlib/library/targets/hwlib-native.hpp]: Is it not better to use 

```cpp
#define OS_Linux
    #include <sys/time.h>
    #define EPOCH_DIFF 11644473600LL
 #elif defined(_WIN32) || defined(WIN32) /* _Win32 is usually defined by compilers targeting 32 or   64 bit Windows systems */
    #define OS_Windows
    #include <windows.h>
 #endif

```

Comment (itzandroidtab) [reply libraries/hwlib/library/targets/hwlib-native.hpp]: The indents on the lines after this one seem a bit out of line as well

**Comment by itzandroidtab** [comment]
It looks like it is working on windows. I'm not sure about Linux sinds I don't have the environment running on a Linux box

File: programs/bmptk/Makefile.inc (hunk 2 above)
Comment (gerritvanos) [reply programs/bmptk/Makefile.inc]: Yes due to some strange thing in HWLIB and bmptk it is not necessary and it still adds the flag running native on linux which gives problems
it is a temporary fix for now

File: libraries/hwlib/library/targets/hwlib-native.hpp (hunk 3 above)
Comment (gerritvanos) [reply libraries/hwlib/library/targets/hwlib-native.hpp]: yes the define could be removed, and will look at the indents

**Comment by gerritvanos** [comment]
Windows check after changes

**Comment by itzandroidtab** [comment]
I changed the rest of the tabs to spaces so the indents look correct on github now

**Comment by gerritvanos** [comment]
linux check aswell now

**Pull request merged**